mos-platform/nes/nesdoug/vram_buffer.c

```diff
@@ -2,12 +2,14 @@
 // version 1.2, 1/1/2022
 
 __attribute__((section(".zp.vram_index"))) char VRAM_INDEX;
+__attribute__((weak, aligned(128), section(".noinit.vram_buf"))) char VRAM_BUF[128];
 
-extern char VRAM_BUF[];
-
-void set_vram_update(const void *buf);
-void set_vram_buffer(void) {
+void __post_vram_update() {
```

Comment:
This must be callable from and interrupt context, and right now C isn't unless it's `__attribute__((interrupt))` or `__attribute__((interrupt_norecurse))`. Without the AXY calling convention, these will blow up the sizes of the routine, so for now, this needs to stay in assembly.

Reply:
Hmm... I was following the example of `__early_init`/`__late_init`. For NMI, I was also under the impression a void(void) function that only does basic memory ops on global variables wouldn't need anything besides A, X, and Y, which are preserved by the normal NMI preamble.

But yeah, having a c function that can't be used as a normal C function is probably less good than just duplicating the code.

Reply:
> Hmm... I was following the example of `__early_init`/`__late_init`. For NMI, I was also under the impression a void(void) function that only does basic memory ops on global variables wouldn't need anything besides A, X, and Y, which are preserved by the normal NMI preamble.

This is likely to be true, but nothing guarantees this, and code generator flak may sneak in imaginary register references. It's best not to depend on the specific output of the code generator for given C; one should instead rely upon ABI guarantees. That's why I'm hoping to eventually make "only touches AXY" an ABI guarantee: so we can rely on it.

Reply:
Hmm... the commit where I moved it into C said it was to fix something. I need to remember what that was 🤔

Reply:
Oh ... `21_sprite_zero` calls `set_vram_update` but doesn't use any of the functions that are defined in `vram_buffer.s`. It only uses `metatile.c/s`. 

So I can either have `set_vram_buffer` call the assembly version of `__post_vram_update` or i can separate `multi/one-vram-buffer` functions from the buffer definition.